modules/Oculus_module/src/FingersRetargeting.cpp

```diff
@@ -7,63 +7,58 @@
  * @date 2018
  */
 
-#include "FingersRetargeting.hpp"
-#include "Utils.hpp"
+#include <FingersRetargeting.hpp>
+#include <Utils.hpp>
 
-bool FingersRetargeting::configure(const yarp::os::Searchable& config)
+bool FingersRetargeting::configure(const yarp::os::Searchable& config, const std::string& name)
 {
-    // TODO do it in a better way
-    m_fingersJoints = 7;
+    m_controlHelper = std::make_unique<RobotControlHelper>();
+    if (!m_controlHelper->configure(config, name))
+    {
+        yError() << "[FingersRetargeting::configure] Unable to configure the finger helper";
+        return false;
+    }
+
+    int fingersJoints = m_controlHelper->getDoFs();
+
     double samplingTime;
     if (!YarpHelper::getDoubleFromSearchable(config, "samplingTime", samplingTime))
     {
         yError() << "[configure] Unable to find the head smoothing time";
         return false;
     }
 
-    m_fingerVelocity.resize(m_fingersJoints);
-    if (!YarpHelper::getYarpVectorFromSearchable(config, "fingersVelocity", m_fingerVelocity))
+    m_fingersScaling.resize(fingersJoints);
+    if (!YarpHelper::getYarpVectorFromSearchable(config, "fingersScaling", m_fingersScaling))
     {
         yError() << "[configure] Initialization failed while reading "
                     "fingersVelocity vector.";
         return false;
     }
 
-    m_desiredPosition.resize(m_fingersJoints);
-    yarp::sig::Vector buff(m_fingersJoints, 0.0);
-    m_fingerIntegrator = std::make_unique<iCub::ctrl::Integrator>(samplingTime, buff);
-    return true;
-}
+    m_desiredJointPosition.resize(fingersJoints);
+    yarp::sig::Vector buff(fingersJoints, 0.0);
+    m_fingerIntegrator = std::make_unique<iCub::ctrl::Integrator>(
+        samplingTime, buff, m_controlHelper->getLimits());
 
-bool FingersRetargeting::closeHand()
-{
-    if (m_fingerIntegrator == nullptr)
+    if (!m_controlHelper->switchToControlMode(VOCAB_CM_POSITION_DIRECT))
     {
-        yError() << "[closeHand] the integrator is not initialized please call "
-                    "configure() method";
+        yError() << "unable to switch the control mode";
```

Comment:
Maybe we can avoid to ``return false`` here since some joints may be in HF. We may not care too much if some joints won't move in this case.